src/cli/config.js

```diff
@@ -2,35 +2,121 @@ import path from "path";
-                    return res(false);
-                }
-                // if not, get next folder
-                getConfigurationFile(
-                    filename,
-                    path.resolve(folder, "..")
-                ).then(res).catch(rej);
+                return res(true);
             }
+            res(false);
         });
     });
 }
 
+/**
+ * Check if a package.json file is defining the property "type" to "module".
+ * @param {String} filepath The package.json file to check
+ * @returns {Boolean} true if `"type": "module"` is defined, false otherwise
+ **/
+function isEsmPackageJson(filename) {
+    try {
+        let pkg = JSON.parse(fs.readFileSync(filename, "utf8"));
+        return pkg.type && pkg.type === "module";
+    } catch (err) {
+        return false;
+    }
+}
+
+/**
+ * Check if the default configuration file for SVGLint exists in a folder
+ * @param {String} folder The folder to check
+ * @returns {Promise<String,Boolean>} The path to the file if it exists, false otherwise
+ */
+async function getDefaultConfigurationFile(folder) {
+    let filepath = path.resolve(folder, ".svglintrc.js");
+    if (await fileExists(filepath)) {
+        return filepath;
+    }
+
+    const packageJsonPath = path.resolve(folder, "package.json");
+    if (await fileExists(packageJsonPath)) {
+        filepath = path.resolve(
+            folder,
+            isEsmPackageJson(packageJsonPath) ? ".svglintrc.cjs" : ".svglintrc.mjs",
+        );
+        if (await fileExists(filepath)) {
+            return filepath;
+        }
+    }
+
+    return false;
+}
+
+/**
+ * Gets the configuration file to use, traversing the parent folders
+ * @param {String} folder The folder to look in
+ * @returns {Promise<String,Boolean>} The path to the configuration file, or false
+ */
+async function getDefaultConfigurationFileTraversingParents(folder) {
+    let filepath = await getDefaultConfigurationFile(folder);
+    if (filepath) {
+        return filepath;
+    } else {
+        const parent = path.resolve(folder, "..");
+        if (parent === folder) {
+            return false;
+        }
+        return await getDefaultConfigurationFileTraversingParents(parent);
+    }
+}
+
+/**
+ * Get the configuration file to use
+ * @param {String} filename The filename to look for
+ * @param {String} folder The folder to look in
+ * @returns {Promise<String,Boolean>} The path to the configuration file, or false
+ */
+async function getConfigurationFile(filename, folder) {
+    let filepath;
+    if (filename) {
+        filepath = path.isAbsolute(filename)
+            ? filename
+            : path.resolve(folder, filename);
+        if (await fileExists(filepath)) {
+            return filepath;
+        } else {
+            return false;
+        }
+    }
+
+    filepath = await getDefaultConfigurationFile(folder);
+    if (filepath) {
+        return filepath;
+    }
+
+    return await getDefaultConfigurationFileTraversingParents(folder);
```

Comment:
If I'm not mistaken this can be simplified to

```suggestion
return await getDefaultConfigurationFileTraversingParents(folder);
```

since

```js
filepath = await getDefaultConfigurationFile(folder);
if (filepath) {
    return filepath;
}
```

is the first thing `getDefaultConfigurationFileTraversingParents` does.